File: CSharpFundamentals/CSharpOOPbasics/Exam 18.11.2018/Factories/AnimalFactory.cs
using AnimalCentre.Models;
using AnimalCentre.Models.Contracts;
using System;
using System.Collections.Generic;
using System.Text;

namespace AnimalCentre.Factories
{
    public class AnimalFactory
    {
        public Animal CreateAnimal(string type, string name, int energy,
            int happiness, int procedureTime)
        {
            Animal animal = null;

            switch (type)
            {
                case "Cat":
                    animal = new Cat(name, energy, happiness, procedureTime);
                    break;
                case "Dog":
                    animal = new Dog(name, energy, happiness, procedureTime);
                    break;
                case "Lion":
                    animal = new Lion(name, energy, happiness, procedureTime);
                    break;
                case "Pig":
                    animal = new Pig(name, energy, happiness, procedureTime);
                    break;
            }

            return animal;
        }
    }
}

File: JavaDB/DatabaseAdvancedFrameworks/JavaOOPPrinciplesEx/Animals/factories/AnimalFactory.java
package Animals.factories;

import Animals.models.*;

import static Animals.constraints.Constraints.FEMALE;
import static Animals.constraints.Constraints.INVALID_INPUT;
import static Animals.constraints.Constraints.MALE;

public class AnimalFactory {
    public Animal CreateAnimal(String type, String name, int age,
                               String gender){
        Animal animal = null;
        switch (type){
            case "Dog":
                animal = new Dog(name, age, gender);
                break;
            case "Cat":
                animal = new Cat(name, age, gender);
                break;
            case "Frog":
                animal = new Frog(name, age, gender);
                break;
            case "Kitten":
                animal = new Kitten(name, age, FEMALE);
                break;
            case "Tomcat":
                animal = new Tomcat(name, age, MALE);
                break;
                default:
                    throw new IllegalArgumentException(INVALID_INPUT);
        }
        return animal;
    }
}
